test/testsimplifytypedef.cpp

```diff
@@ -2544,6 +2545,63 @@ class TestSimplifyTypedef : public TestFixture {
         ASSERT_EQUALS(exp, tok(code, false));
     }
 
+    void simplifyTypedef129() { 
+        {
+           const char code[] = "class c {\n"
+                              "  typedef char foo[4];\n"
+                              "  foo &f ;\n"
+                              "};";
+
+            const char exp [] = "class c { char ( & f ) [ 4 ] ; } ;";
+            ASSERT_EQUALS(exp, tok(code, false));
+        }
+
+        {
+           const char code[] = "class c {\n"
+                              "  typedef char foo[4];\n"
+                              "  const foo &f;\n"
+                              "};";
+
+            const char exp [] = "class c { const char ( & f ) [ 4 ] ; } ;";
+            ASSERT_EQUALS(exp, tok(code, false));
+        }
+
+        {
+           const char code[] = "class c {\n"
+                              "  typedef char foo[4];\n"
+                              "  foo _a;\n"
+                              "  constexpr const foo &c_str() const noexcept { return _a; }\n"
+                              "};";
+
+            const char exp [] = "class c { char _a [ 4 ] ; const const char ( & c_str ( ) const noexcept ) [ 4 ] { return _a ; } } ;";
```

Comment:
I guess this is the correct expansion.. however I admit I don't understand this simplification. I'll trust you on this.

However I guess that for instance `isFunctionHead` won't handle this code well.. and then function handling does not work. If you run Cppcheck like `cppcheck --debug -v 2211.cpp` on this code then we can see that the function c_str() is not printed out. I wonder... do you want to look into this further later? I would suggest you do that in a separate PR.